src/decode-ethernet.c

```diff
@@ -38,6 +38,8 @@
 #include "util-unittest.h"
 #include "util-debug.h"
 
+#define MAX_ETH_OFFSET 256
```

Comment:
This is a simple fix.

Another fix could be to count recursion levels in every `DecodeX` function

A better fix would to have a generic `DecodeFirstLayers` function which will loop calling every `DecodeX` function, and these `DecodeX` functions should now return the number of consumed bytes, and the next layer to decode... So this would be a deeper bigger change